@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -313,46 +345,118 @@ export function placePointLabel(
 
     if (label.bounds === undefined) {
         label.bounds = new THREE.Box2();
-        tmpMeasurementParams.outputCharacterBounds = undefined;
-        tmpMeasurementParams.path = undefined;
-        tmpMeasurementParams.pathOverflow = false;
-        tmpMeasurementParams.letterCaseArray = label.glyphCaseArray!;
-        textCanvas.measureText(label.glyphs!, label.bounds, tmpMeasurementParams);
     }
 
-    screenPosition.add(computePointTextOffset(label, scale, tmpTextOffset));
-    outScreenPosition.set(screenPosition.x, screenPosition.y, labelState.renderDistance);
-
-    // TODO: Make the margin configurable
-    tmpBox.copy(label.bounds!).expandByVector(pointLabelMargin);
-    tmpBox.min.multiplyScalar(scale);
-    tmpBox.max.multiplyScalar(scale);
-    tmpBox.translate(screenPosition);
-    tmp2DBox.set(
-        tmpBox.min.x,
-        tmpBox.min.y,
-        tmpBox.max.x - tmpBox.min.x,
-        tmpBox.max.y - tmpBox.min.y
-    );
-
-    // Check the text visibility.
-    if (!screenCollisions.isVisible(tmp2DBox)) {
-        return PlacementResult.Invisible;
-    }
+    // Setup measurements parameters for textCanvas.measureText().
+    tmpMeasurementParams.outputCharacterBounds = undefined;
+    tmpMeasurementParams.path = undefined;
+    tmpMeasurementParams.pathOverflow = false;
+    tmpMeasurementParams.letterCaseArray = label.glyphCaseArray!;
+
+    // NOTE:
+    // We may save some CPU time by firstly checking visibility of all possible text placements,
+    // their bounding boxes merged altogether, this may save time for further calculations.
+    // Would be enough to include TL and BR placements.
+
+    // Start with currently set alignment settings.
+    const basePlacement = {
```

Comment:
if the label is a persistent one, the basePlacement is not the one given in the style, but the one which is currently used by the label.
This means you're gonna have to store the current placement somewhere.

Reply:
Yes, Andres and this is actually stored in label.layoutStyle!.horizontal/verticalAlignment, as far as I investigated it, this objects are not removed, so it will be hold until style is changed (zoom level change, etc..).  That is why I write directly to it... moreover textCanvas keeps shallow copy (reference) of label layout thus it is used internally everywhere during the measurement and text buffer creation. This was the main reason I couldn't pass new placement as the parameter, as tried in `computePointTextOffset` method. The change to fix that was massive and I have surrendered.... :).

Reply:
> if the label is a persistent one, the basePlacement is not the one given in the style, but the one which is currently used by the label.

As previously said - this is intentional. If we would like to store the new placement in the TextElementState, and leave TextElement.layoutStyle untouched, then we would need to override TextCanvas.textLayoutStyle anyway. Moreover for persistent labels this would need to be done all the time (with constant searching for better placement), because original layout will not fit in most cases (if camera does not change too much).

Reply:
Current placement is now stored in TextElementState.LayoutState.